lib/IRGen/GenType.cpp

```diff
@@ -2937,20 +2938,34 @@ static bool tryEmitDeinitCall(IRGenFunction &IGF,
     return true;
   }
 
+  auto deinitSILFn = deinitTable->getImplementation();
+
+  // Look for a specialization of deinit that we can call.
+  auto substitutions = ty->getContextSubstitutionMap();
+  if (!substitutions.empty() &&
+      !substitutions.getRecursiveProperties().hasArchetype()) {
+    Mangle::GenericSpecializationMangler mangler(
```

Comment:
This is kind of gross, is there a better way where we actually store the specializations in a table somewhere?

Reply:
It is totally gross, I agree. A better answer I was considering is to allow `SILMoveOnlyDeinit` to carry a substitution map, and make it possible to look up `(nominal type, substitution map)` pairs in the module + serialized module. We would specialize the `SILMoveOnlyDeinit` when we see a nominal type specialization that has a `deinit`.

Reply:
I'm going to start with my hack to unblock some stuff